### PR TITLE
Consolidate all dependencies in Versions.props

### PR DESCRIPTION
**Pull request by ericstj**
Can be reviewed commit-by-commit.

1. Organized and removed unused properties.
2. Applied the same naming convention of packageName.Replace(".", "") + "Version" for all properties
3. Moved all static PackageReference/@Version from project files to central props.

I expect this change to largely be no-difference with one small exception, noted below.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -12,126 +12,76 @@
...
+    <MicrosoftExtensionsVersion>2.1.0</MicrosoftExtensionsVersion>
+    <SystemBuffersVersion>4.5.0</SystemBuffersVersion>
+    <SystemCodeDomVersion>4.5.0</SystemCodeDomVersion>
+    <SystemCollectionsImmutableVersion>1.5.0</SystemCollectionsImmutableVersion>
+    <SystemDrawingCommonVersion>4.5.0</SystemDrawingCommonVersion>
     <SystemIOFileSystemAccessControl>4.5.0</SystemIOFileSystemAccessControl>
+    <SystemMemoryVersion>4.5.3</SystemMemoryVersion>
+    <SystemReflectionEmitLightweightVersion>4.3.0</SystemReflectionEmitLightweightVersion>
+    <SystemReflectionEmitVersion>4.3.0</SystemReflectionEmitVersion>
+    <SystemRuntimeCompilerServicesUnsafeVersion>6.0.0</SystemRuntimeCompilerServicesUnsafeVersion>
```
Comment (ericstj) [inline eng/Versions.props:27]: This results in a change from `4.7.0` to `6.0.0` in https://github.com/dotnet/machinelearning/blob/04dda55ab0902982b16309c8e151f13a53e9366d/src/Microsoft.Data.Analysis/Microsoft.Data.Analysis.csproj#L48

That should be fine and I will follow up with further package updates to all dependencies.

File: eng/Versions.props [hunk 2]
```diff
@@ -12,127 +12,77 @@
...
     <MicrosoftSourceLinkVersion>1.1.0-beta-20206-02</MicrosoftSourceLinkVersion>
-    <BuildToolsPackageVersion>3.0.0-preview4-04926-01</BuildToolsPackageVersion>
+    <MicrosoftSymbolUploaderBuildTaskVersion>1.1.225302</MicrosoftSymbolUploaderBuildTaskVersion>
+    <ReportGeneratorVersion>4.3.6</ReportGeneratorVersion>
+    <SystemCompositionVersion>1.2.0</SystemCompositionVersion>
+
     <!-- Test-only Dependencies -->
+    <ApprovalTestsVersion>5.4.7</ApprovalTestsVersion>
     <BenchmarkDotNetVersion>0.12.0</BenchmarkDotNetVersion>
+    <DotNetRuntime21Version>2.1.0</DotNetRuntime21Version>
```
Comment (michaelgsharp) [inline eng/Versions.props:69]: Do we even need to keep this one anymore?

File: eng/helix.proj [hunk 3]
```diff
@@ -39,13 +39,13 @@
...
   <ItemGroup>
-    <AdditionalDotNetPackage Include="$(MicrosoftNETCore3PlatformsVersion)" Condition="'$(BuildArchitecture)' != 'arm64' Or !$(HelixTargetQueues.ToLowerInvariant().Contains('osx'))">
+    <AdditionalDotNetPackage Include="$(DotNetRuntime31Version)" Condition="'$(BuildArchitecture)' != 'arm64' Or !$(HelixTargetQueues.ToLowerInvariant().Contains('osx'))">
       <PackageType>runtime</PackageType>
       <DotNetCliRuntime Condition="'$(BuildArchitecture)' == 'x86'">win-x86</DotNetCliRuntime>
       <DotNetCliRuntime Condition="'$(BuildArchitecture)' == 'arm' And $(HelixTargetQueues.ToLowerInvariant().Contains('ubuntu'))">linux-arm</DotNetCliRuntime>
     </AdditionalDotNetPackage>
 
-    <AdditionalDotNetPackage Include="$(MicrosoftNETCorePlatformsVersion)" Condition="'$(BuildArchitecture)' != 'arm64' Or !$(HelixTargetQueues.ToLowerInvariant().Contains('osx'))">
+    <AdditionalDotNetPackage Include="$(DotNetRuntime21Version)" Condition="'$(BuildArchitecture)' != 'arm64' Or !$(HelixTargetQueues.ToLowerInvariant().Contains('osx'))">
```
Comment (michaelgsharp) [inline eng/helix.proj:48]: IF we don't need to keep the 21Version in props, we can remove this here.

File: eng/Versions.props [hunk 4]
```diff
@@ -12,127 +12,77 @@
...
-    <SystemThreadingChannelsPackageVersion>4.7.1</SystemThreadingChannelsPackageVersion>
-    <!-- Other/External dependencies -->
-    <GoogleProtobufPackageVersion>3.19.4</GoogleProtobufPackageVersion>
-    <LightGBMPackageVersion>2.3.1</LightGBMPackageVersion>
-    <MicrosoftExtensionsPackageVersion>2.1.0</MicrosoftExtensionsPackageVersion>
-    <MicrosoftMLOnnxRuntimePackageVersion>1.10.0</MicrosoftMLOnnxRuntimePackageVersion>
-    <MlNetMklDepsPackageVersion>0.0.0.12</MlNetMklDepsPackageVersion>
-    <ParquetDotNetPackageVersion>2.1.3</ParquetDotNetPackageVersion>
-    <SystemDrawingCommonPackageVersion>4.5.0</SystemDrawingCommonPackageVersion>
+
```
Comment (tannergooding) [inline eng/Versions.props:15]: More generally, have we considered switching to the official NuGet support for managing versions?

https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-packages

File: eng/Versions.props (hunk 2 above)
Comment (ericstj) [reply eng/Versions.props:69]: I can see about removing it in a follow up PR.  This PR is just organization.

File: eng/Versions.props (hunk 4 above)
Comment (ericstj) [reply eng/Versions.props:15]: The current patterns are established by Arcade.  We could see about other patterns to follow moving forward, but right now we're taking some incremental steps.

Comment (tannergooding) [reply eng/Versions.props:15]: Logged https://github.com/dotnet/arcade/issues/9153

**Review by michaelgsharp — APPROVED** [review]

File: eng/Versions.props (hunk 4 above)
Comment (ericstj) [reply eng/Versions.props:15]: Thanks @tannergooding

**Pull request merged**